connectors/sources/network_drive.py

```diff
@@ -387,13 +402,18 @@ async def find_matching_paths(self, advanced_rules):
 
     async def ping(self):
         """Verify the connection with Network Drive"""
+
+        if self.smb_connection.session is not None:
+            return
```

Comment:
Why? We still need to be able to ping the backend.

Reply:
Apologies, I was trying multiple approaches and somehow missed to remove the singleton approach.